lib/apia/response.rb

```diff
@@ -53,11 +63,20 @@ def body
       @body || hash
     end
 
+    def type
+      @type || JSON
+    end
+
     # Return the rack triplet for this response
     #
     # @return [Array]
     def rack_triplet
-      Rack.json_triplet(body, headers: @headers, status: @status)
+      case type
+      when JSON
+        Rack.json_triplet(body, headers: headers, status: status)
+      when PLAIN
+        Rack.plain_triplet(body, headers: headers, status: status)
+      end
```

Comment:
Nitpick: Since we're in the area, and we have the `Rack.response_triplet` helper method now, do you think it might be worth expanding the response object slightly to also allow fully custom response body objects and content types?

This might also have even further implications with the OpenAPI schema though. Basically, if this wouldn't be a quick win, let's ignore it until we need it :)